### DISCUSSION TITLE
Recommended cluster specs and spark configuration for larger datasets (Databricks)?

### DISCUSSION THREAD

**Discussion by rjc89** [moj-analytical-services/splink#1580] [Q&A] [answered]
Hi, what's the view on cluster specs and spark configuration options recommended for tackling large linkage jobs (50-100 million records) with Splink these days? I'm using Databricks and have access to various cluster options but obviously costs can mount or you can encounter out of resource errors if your cluster's are not appropriately sized and configured. Cheers

**Comment by RobinL** [comment, marked as answer]
Sorry for the delay.

Our linkage jobs at that scale i believe have a cluster size (on AWS Glue) of around 10.  Note that the blocking rules between them generate I think a few hundred million comparisons.

Things to look out for/try:
- Skew in your data.  For instance, if you block on full name, that might be find for Robin Linacre, but is likely to create a huge number of comparisons for John Smith
- Set `spark.default.parallelism` fairly high, try 1000
- I recommend quite a long list of blocking rules for prediction, that individually are quite strict (try to keep below about 1bn total comparisons to begin with) see also [here](https://moj-analytical-services.github.io/splink/demos/tutorials/03_Blocking.html#devising-effective-blocking-rules)
- Repartition data prior to loading into Splink (probably not necessary but can help if you have e.g. 1 very large input file)

If you're still struggling, you could try using [less computationally intensive settings](https://github.com/moj-analytical-services/splink/discussions/1365).
